tests/phpunit/test-class-wc-stripe-upe-payment-method.php

```diff
@@ -0,0 +1,319 @@
+<?php
+/**
+ * Unit tests for UPE payment methods
+ */
+class WC_Stripe_UPE_Payment_Method_Test extends WP_UnitTestCase {
+	/**
+	 * Array of mocked UPE payment methods.
+	 *
+	 * @var array
+	 */
+	private $mock_payment_methods = [];
+
+	/**
+	 * Mock capabilities object from Stripe response--all inactive.
+	 */
+	const MOCK_INACTIVE_CAPABILITIES_RESPONSE = [
+		'bancontact_payments' => 'inactive',
+		'card_payments'       => 'inactive',
+		'eps_payments'        => 'inactive',
+		'giropay_payments'    => 'inactive',
+		'ideal_payments'      => 'inactive',
+		'p24_payments'        => 'inactive',
+		'sepa_debit_payments' => 'inactive',
+		'sofort_payments'     => 'inactive',
+		'transfers'           => 'inactive',
+	];
+
+	/**
+	 * Mock capabilities object from Stripe response--all active.
+	 */
+	const MOCK_ACTIVE_CAPABILITIES_RESPONSE = [
+		'bancontact_payments' => 'active',
+		'card_payments'       => 'active',
+		'eps_payments'        => 'active',
+		'giropay_payments'    => 'active',
+		'ideal_payments'      => 'active',
+		'p24_payments'        => 'active',
+		'sepa_debit_payments' => 'active',
+		'sofort_payments'     => 'active',
+		'transfers'           => 'active',
+	];
+
+	/**
+	 * Initial setup
+	 */
+	public function setUp() {
+		parent::setUp();
+		$this->reset_payment_method_mocks();
+	}
+
+	/**
+	 * Reset mock_payment_methods to array of mocked payment methods
+	 * with no mocked expectations for methods.
+	 */
+	private function reset_payment_method_mocks() {
+		$this->mock_payment_methods = [];
+		foreach ( WC_Stripe_UPE_Payment_Gateway::UPE_AVAILABLE_METHODS as $payment_method_class ) {
+			$mocked_payment_method = $this->getMockBuilder( $payment_method_class )
+				->setMethods(
+					[
+						'get_capabilities_response',
+						'get_woocommerce_currency',
+						'is_subscription_item_in_cart',
+					]
+				)
+				->getMock();
+			$this->mock_payment_methods[ $mocked_payment_method->get_id() ] = $mocked_payment_method;
+		}
+	}
```

Comment:
This necessary so that I can set different return values for these functions inside the same test. There are other ways to do this (e.g. I could replace `$this->any()` with `$this->exactly( n )` and then list out _all_ the expected return values in order), but this seemed the cleanest.